packages/jira-adapter/src/filters/automation/automation_structure.ts

```diff
@@ -265,7 +288,31 @@ const revertCompareFieldValueStructure = async (instance: InstanceElement): Prom
   })).value
 }
 
-const filter: FilterCreator = () => {
+const isOldFormatScope = (projects: Values): boolean =>
+  Array.isArray(projects) && projects.length > 0
+
+const convertRuleScopeToProjects = (instance: InstanceElement): void => {
+  const { ruleScope } = instance.value
+  if (!isRuleScope(ruleScope) || isOldFormatScope(instance.value.projects)) {
+    return
+  }
+  instance.value.projects = []
+  ruleScope.resources.forEach((resource: string) => {
+    if (resource.match(PROJECT_SCOPE_REGEX)) {
+      instance.value.projects.push({ projectId: resource.match(PROJECT_SCOPE_REGEX)?.[1] })
```

Comment:
I think you can avoid the `?` by doing 
```suggestion
    const projectScopeMatch = resource.match(PROJECT_SCOPE_REGEX)
    if (projectScopeMatch) {
      instance.value.projects.push({ projectId: projectScopeMatch[1] })
```

same for projectTypeKey

Reply:
I was thinking about it, but with the if-else it required doing all 3 matches before the if-else and it looked less clean to me

Reply:
I think if you extract it to a function like in https://github.com/salto-io/salto/pull/4508#discussion_r1230542244 you won't need the if-else and you can do
```
const match1 = ..
if (match1) {
  return ...
}
const match2 = ..
if (match2) {
  return ...
}
...
```